ui/src/app/services/fetch/fetch.service.ts

```diff
@@ -103,6 +103,10 @@ export class FetchService {
     })
   }
 
+  setGcsDetailsForShardedMigrations(payload: IGcsConfig) {
+    return this.http.post(`${this.url}/SetGcsDetailsForShardedMigrations`, payload)
+  }
+
   setDataflowDetailsForShardedMigrations(payload: IDataflowConfig) {
     return this.http.post(`${this.url}/SetDataflowDetailsForShardedMigrations`, {
       DataflowConfig: payload
```

Comment:
We should think about refactoring this to create a generic API.

Reply:
We can probably have a common method definition but we will need some switch cases inside backend to differentiate the resources.

Its 3 small methods vs 1 large method.

Created this for tracking anyways: b/313957500